hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -58,22 +59,27 @@
   private ThreadLocal<BinaryDecoder> decoderCache = new ThreadLocal<>();
 
   public HoodieAvroDataBlock(@Nonnull Map<HeaderMetadataType, String> logBlockHeader,
-       @Nonnull Map<HeaderMetadataType, String> logBlockFooter,
-       @Nonnull Option<HoodieLogBlockContentLocation> blockContentLocation, @Nonnull Option<byte[]> content,
-       FSDataInputStream inputStream, boolean readBlockLazily) {
+                             @Nonnull Map<HeaderMetadataType, String> logBlockFooter,
+                             @Nonnull Option<HoodieLogBlockContentLocation> blockContentLocation, @Nonnull Option<byte[]> content,
+                             FSDataInputStream inputStream, boolean readBlockLazily) {
     super(logBlockHeader, logBlockFooter, blockContentLocation, content, inputStream, readBlockLazily);
   }
 
   public HoodieAvroDataBlock(HoodieLogFile logFile, FSDataInputStream inputStream, Option<byte[]> content,
-       boolean readBlockLazily, long position, long blockSize, long blockEndpos, Schema readerSchema,
-       Map<HeaderMetadataType, String> header, Map<HeaderMetadataType, String> footer) {
+                             boolean readBlockLazily, long position, long blockSize, long blockEndpos, Schema readerSchema,
+                             Map<HeaderMetadataType, String> header, Map<HeaderMetadataType, String> footer, String keyField) {
```

Comment:
looks like keyField is not used in AvroDataBlock and is used only in HFileDataBlock. is it possible to confine the usage to just HFile data blocks if it makes sense?

Reply:
HoodieAvroBlock is derived from HoodieDataBlock and making sure the user provided key fields are sent to the parent classes properly.